src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -533,10 +523,13 @@ private static bool ContainsSpaceOrAsterisk(string text)
             //      candidate in a case *sensitive* manner. If so, return that there was a substring
             //      match.
             //
-            //   e) If the word was not entirely lowercase, then attempt a camel cased match as
-            //      well.
+            //   e) If the word was entirely lowercase, then attempt a special lower cased camel cased 
```

Comment:
For example, *CofiPro* won't match *CodeFixProvider* even with the new implementation, but if the new mode were made case insensitive then it would.